test/integration/ResourceLockCleanup.test.ts

```diff
@@ -0,0 +1,128 @@
+import fetch from 'cross-fetch';
+import type { App, DataAccessorBasedStore, FileSystemResourceLocker, ReadWriteLocker } from '../../src';
+import { readableToString, BasicRepresentation } from '../../src';
+import { describeIf, getPort } from '../util/Util';
+import { getDefaultVariables, getTestConfigPath, getTestFolder, instantiateFromConfig, removeFolder } from './Config';
+
+const port = getPort('ResourceLockCleanup');
+const baseUrl = `http://localhost:${port}/`;
+
+describe('A server using locking', (): void => {
+  const resourceIdentifier = { path: `${baseUrl}container1/test.txt` };
+
+  describe('in case of file-based locking', (): void => {
+    let app: App;
+    let store: DataAccessorBasedStore;
+    let locker: FileSystemResourceLocker;
+    const rootFilePath = getTestFolder(`resource-lock-cleanup`);
+
+    beforeAll(async(): Promise<void> => {
+      // Clean the working dir
+      await removeFolder(rootFilePath);
```

Comment:
Good point! I sometimes terminate the process when debugging, which can lead to folder content remaining on disk. I will remove this statement, as this will never occur in the CI pipeline.